### PR TITLE
For #17797: Add bookmarks menu item.

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by eliserichards** [comment]
Is this ready for review? 😄

**Comment by mcarare** [comment]
@elis It should be now. :)

File: app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt [hunk 1]
```diff
@@ -495,12 +487,29 @@ class DefaultToolbarMenu(
...
+            imageResource = R.drawable.ic_bookmarks_menu,
+            iconTintColorResource = primaryTextColor(),
+            label = context.getString(R.string.library_bookmarks),
+            labelListener = {
+                onItemTapped.invoke(ToolbarMenu.Item.Bookmarks)
+            },
+            primaryStateIconResource = R.drawable.ic_bookmark_outline,
+            secondaryStateIconResource = R.drawable.ic_bookmark_filled,
+            tintColorResource = accentBrightTextColor(),
+            primaryLabel = "Add",
```
[the remark this answers is not in the document]
Comment (mcarare) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt:500]: Yes, you are right, I was using this for testing the AC item and forgot to change this.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -7,6 +7,10 @@
     <string name="cancel">Cancel</string>
     <!-- A 'Save' string to be used for labeling UI elements. -->
     <string name="save">Save</string>
+    <!-- An 'Add' string to be used for labeling UI elements. -->
+    <string name="add_label">Add</string>
```
Comment (gabrielluong) [inline app/src/main/res/values/strings.xml:11]: Curious if we can just name these "add" and "edit" for the string name?

File: app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt [hunk 3]
```diff
@@ -524,10 +533,19 @@ class DefaultToolbarMenu(
         menuItems
     }
 
+    private fun bookmarkListener() {
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt:536]: Maybe some variation of our handleX naming.

```suggestion
    private fun handleBookmarkItemTapped() {
```

**Review by gabrielluong — APPROVED** [review]

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (mcarare) [reply app/src/main/res/values/strings.xml:11]: @gabrielluong There already is a string for saved logins named just "edit" and I wanted to have a separate resource from that one.

Comment (gabrielluong) [reply app/src/main/res/values/strings.xml:11]: @mcarare oh i see. I am wondering now if we should've just used that one and moved it in with the generic string. I did a quick search for `R.string.edit` and didn't see it being used except in a snackbar test, which makes it generic enough.

**Pull request merged**